codemod/base.py

```diff
@@ -396,7 +407,12 @@ def generate_patches(self):
         start_pos = self.start_position or Position(None, None)
         end_pos = self.end_position or Position(None, None)
 
-        path_list = Query._walk_directory(self.root_directory)
+        path_list = None
+        if self._use_spotlight:
```

Comment:
On the topic of using a single `spotlight_match_pattern` parameter, this could be:

```py
if self._spotlight_match_pattern is not None:
    # ...
```